.gitlab-ci.yml

```diff
@@ -10,13 +10,26 @@ stages:
   - deploy_dev
   - deploy_prod
 
-build:
+build_dev:
   stage: build
   tags:
     - dind
   script:
     - docker login -u gitlab-ci-token -p $CI_JOB_TOKEN $CI_REGISTRY
-    - docker build -t $IMAGE_NAME .
+    - docker build -t $IMAGE_NAME --build-arg DEPLOYMENT_ENV=staging .
+    - docker push $IMAGE_NAME
+
+build_prod:
+  stage: build
+  tags:
+    - dind
+  only:
+    - tags
+  script:
+    - docker login -u gitlab-ci-token -p $CI_JOB_TOKEN $CI_REGISTRY
+    - export DEV_IMAGE_NAME=$CI_REGISTRY_IMAGE/master:$CI_COMMIT_SHA
+    - docker pull $DEV_IMAGE_NAME || true
+    - docker build --cache-from $DEV_IMAGE_NAME -t $IMAGE_NAME --build-arg DEPLOYMENT_ENV=prod .
     - docker push $IMAGE_NAME
```

Comment:
@aaclan-ebi Not sure how effective is `--cache-from`, documentation says `Change in one layer causes all subsequent layers to be recreated.`. So in our case code is always new, so we don't want cache and if we go by above statement docker image always going to get recreated since angular build meta data changing.

Other images like ingest or Nginx they will be already cached, it will not download everytime.

Reply:
If I have to optimize build then I will try to reduce angular build time.

Reply:
Thanks @sandykadam ! That's a very good point. I'll push it for now to unblock other user story tickets. But, I'll look into your suggestion!